File: chapter03_2/Ex3_2.java
package chapter03_2;

import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class Ex3_2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int k = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        Arrays.sort(arr);

        int fist = arr[n-1];
        int second = arr[n-2];

        int sum = 0;

        int num_f = ( m / (k + 1) ) * k;
        num_f += m % (k +1);

        sum += fist * num_f + second * ( m / (k+1));
        System.out.println(sum);
    }


}

File: chapter03/Ex3_2.py
n, m, k = map(int, input().split())
inputArray = list(map(int, input().split()))

inputArray.sort()

first = inputArray[n-1]
second = inputArray[n-2]

num_f = int(m / (k+1)) * k
num_f += m % (k+1)

sum = 0
sum += num_f * first 
sum += second * (m//(k+1))

print(sum)